examples/pub_sub_example.rs

```diff
@@ -0,0 +1,165 @@
+use anyhow::Error;
+use env_logger::Env;
+use std::time::{Duration, Instant};
+use tonari_actor::{Actor, Context, System};
+
+#[derive(Debug)]
+enum PublisherMessage {
+    Periodic,
+    Text(String),
+}
+
+impl From<String> for PublisherMessage {
+    fn from(text: String) -> Self {
+        PublisherMessage::Text(text)
+    }
+}
+
+struct PublisherActor {
+    counter: usize,
+    started_at: Instant,
+}
+
+impl PublisherActor {
+    pub fn new() -> Self {
+        Self { counter: 0, started_at: Instant::now() }
+    }
+}
+
+impl Actor for PublisherActor {
+    type Context = Context<Self::Message>;
+    type Error = Error;
+    type Message = PublisherMessage;
+
+    fn name() -> &'static str {
```

Comment:
Off-topic, but I wonder if we can have a default implementation of this method using [type_name](https://doc.rust-lang.org/std/any/fn.type_name.html), since it's mostly what we're manually doing anyway.

Reply:
@PabloMansanet good idea, I've filed #68; resolving this thread.